File: ts/wam/6556.ts
export const WamChannelsVideoPlay = 6556 as const

export interface ChannelsVideoPlayEvent {
  /** field 1, wire `auto_play_t` */
  autoPlayT?: number
  /** field 2, wire `cid` */
  cid?: string
  /** field 3, wire `height` */
  height?: number
  /** field 4, wire `post_id` */
  postId?: string
  /** field 5, wire `video_duration` */
  videoDuration?: number
  /** field 6, wire `video_initial_buffering_t` */
  videoInitialBufferingT?: string
  /** field 7, wire `video_play_origin` */
  videoPlayOrigin?: VIDEOPLAYORIGIN
  /** field 8, wire `video_play_result` */
  videoPlayResult?: VIDEOPLAYRESULT
  /** field 9, wire `video_play_t` */
  videoPlayT?: number
  /** field 10, wire `video_play_type` */
  videoPlayType?: VIDEOPLAYTYPE
  /** field 11, wire `video_size` */
  videoSize?: string
  /** field 12, wire `width` */
  width?: number
  /** field 13, wire `finish_count` */
  finishCount?: number
  /** field 14, wire `watching_module` */
  watchingModule?: string
}

File: rust/wam/6556.rs
pub const WAM_CHANNELS_VIDEO_PLAY: u32 = 6556;

#[derive(Debug, Default)]
pub struct ChannelsVideoPlayEvent {
    /// field 1, wire `auto_play_t`
    pub auto_play_t: Option<i64>,
    /// field 2, wire `cid`
    pub cid: Option<String>,
    /// field 3, wire `height`
    pub height: Option<i64>,
    /// field 4, wire `post_id`
    pub post_id: Option<String>,
    /// field 5, wire `video_duration`
    pub video_duration: Option<i64>,
    /// field 6, wire `video_initial_buffering_t`
    pub video_initial_buffering_t: Option<String>,
    /// field 7, wire `video_play_origin`
    pub video_play_origin: Option<VIDEOPLAYORIGIN>,
    /// field 8, wire `video_play_result`
    pub video_play_result: Option<VIDEOPLAYRESULT>,
    /// field 9, wire `video_play_t`
    pub video_play_t: Option<i64>,
    /// field 10, wire `video_play_type`
    pub video_play_type: Option<VIDEOPLAYTYPE>,
    /// field 11, wire `video_size`
    pub video_size: Option<String>,
    /// field 12, wire `width`
    pub width: Option<i64>,
    /// field 13, wire `finish_count`
    pub finish_count: Option<i64>,
    /// field 14, wire `watching_module`
    pub watching_module: Option<String>,
}